Update api structure (#671)
nervosnetwork/docs.nervos.org: website/docs/script/js/02-05-js-api-ipc.mdx

> (While using JavaScript as a server is technically supported, the overhead of `ckb-js-vm` is significant, so we do **not** recommend doing so.)

Reviewer: ```suggestion
:::warning
Although using JavaScript as a server is technically supported, the overhead of `ckb-js-vm` is significant, so it is NOT recommended.
:::
```

## spawnCellServer

Spawns a new IPC server process using the provided code hash and hash type.

```ts
export function spawnCellServer(
  codeHash: ArrayBuffer,
  hashType: number,
  argv: string[]
): [Pipe, Pipe];
```

**Parameters**

- `codeHash`: The code hash to search for (must be 32 bytes)
- `hashType`: The type of hash to search for (type or data)
- `argv`: An array of strings representing the arguments to pass to the new process.

**Return**

- `[Pipe, Pipe]`: A tuple of `Pipe` objects representing the read and write pipes for the parent process.

**Error**

- Throws on system errors.

## spawnServer

> Same to spawnCellServer, but spawn from a specified source

Reviewer: ```suggestion
Similar to `spawnCellServer`, but spawns a process from a specified source.
```

export function spawnServer(
  index: number,
  source: SourceType,
  argv: string[]
): [Pipe, Pipe] {}
```

**Parameters**

- `index` : The index of the source to spawn from
- `source` : The type of source (use SOURCE\_\* constants)
- `argv`: An array of strings representing the arguments to pass to the new process.

**Return**

- `[Pipe, Pipe]`: A tuple of `Pipe` objects representing the read and write pipes for the parent process.

**Error**

- Throws on system errors.

## runServer

Runs the server with the provided service implementation.
This function listens for incoming requests, processes them using the provided service, and sends back the responses. It uses the inherited file descriptors for communication.

```ts
export function runServer(handler: RequestHandler): void;
```

**Parameters**

- `handler` : The service implementation that handles the requests and generates the responses.

**Error**

- Throws on system errors.

## Channel

The `Channel` class facilitates IPC communication between a client and a server.
It handles the transmission of requests from the client to the server and the reception of responses from the server to the client.

### Channel.constructor

> Creates a new Channel instance

Reviewer: ```suggestion
Creates a new `Channel` instance
```

```ts
constructor(reader: Read, writer: Write);
```

**Parameters**

- `reader` : Responsible for reading data from the channel
- `writer` : Responsible for writing data to the channel

### Channel.execute

Executes the server loop, processing incoming requests and sending responses.

```ts
execute(handler: RequestHandler): void
```

**Parameters**

- `handler` : A service that handles requests and generates responses

### Channel.call

> Sends a request to the server and waits for a response.

Reviewer: ```suggestion
Sends a request to the server and waits for the response.
```
